pkg/querier/blocks_store_queryable.go

```diff
@@ -718,6 +718,10 @@ func (q *blocksStoreQuerier) fetchSeriesFromStores(
 
 			stream, err := c.Series(gCtx, req)
 			if err != nil {
+				if errors.Is(err, context.Canceled) || errors.Is(err, context.DeadlineExceeded) {
```

Comment:
Since this is a stream, would it also make sense to handle the cancellations when receiving from the stream as well?

https://github.com/grafana/mimir/blob/ddf6d54290fa82477eaa5d74012baf66b9bc8e94/pkg/querier/blocks_store_queryable.go#L741

Reply:
Should be covered by the line above:
https://github.com/grafana/mimir/blob/ddf6d54290fa82477eaa5d74012baf66b9bc8e94/pkg/querier/blocks_store_queryable.go#L735-L739

Reply:
thinking about the store-gateway streaming implementation - most of the time in the sotre-gateway is spent waiting to load series and chunks. And much less sending to the querier. To me this means that most of the time in the querier will be spent waiting for a store-gateway response. So the context cancellation is more likely to occur while the querier is waiting on a response from the store-gateway. 

unless i'm missing something i can open a PR for this

Reply:
>So the context cancellation is more likely to occur while the querier is waiting on a response from the store-gateway.

True, but you will still catch it right after the next message is received from gRPC.

Reply:
i don't understand. I thought `srv.Recv()` will return a `contect.Cancelled` error here and we will just swallow it
https://github.com/grafana/mimir/blob/6fca816492917d2f77a9062328890279fda5dcb9/pkg/querier/blocks_store_queryable.go#L741-L748

Reply:
Now I see what you mean and you're right. I misunderstood your previous message. We need to fix it and happy if you can work on it, as you proposed above.

Reply:
Yes, you're definitely right @dimitarvdimitrov, good catch.